### PR TITLE
Bugfix/cc 18 user login status expire after one week

### PR DESCRIPTION
**Pull request by jennawan**
image
When the user refreshes the page after the expiration period (one week), the "UserInfo" with tokens information in the local storage will be removed and the "Sign out" button will become the anonymous avatar. The same scenario for google login, email login and email login then connect with google.

**Hint for testing**: modify the expiresIn value of back-end, CC-server PR CC-18, @src/auth/auth.service.ts line **264** and **269** to a shorter time.
```
    const atExp = {
      secret: process.env.ACCESS_TOKEN_SECRET,
      expiresIn: 60 * 15, //15 min expiration
    };

    const rtExp = {
      secret: process.env.REFRESH_TOKEN_SECRET,
      expiresIn: 60 * 60 * 24 * 7, //one week expiration
    };
```

### CODE REVIEW

**Review by zhuocun — APPROVED** [review]
Great job!

File: src/components/Login/helpers/TokenService.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    if (typeof window !== "undefined") {
+      window.localStorage.removeItem("UserInfo");
+    }
+  }
+
+  setUserToken(userToken: IUserToken | object) {
+    if (typeof window !== "undefined") {
+      const userInfo = JSON.parse(window.localStorage.getItem("UserInfo") as string);
+      userInfo.tokens = userToken;
+      window.localStorage.setItem("UserInfo", JSON.stringify(userInfo));
```
Comment (tatoMa) [inline src/components/Login/helpers/TokenService.ts:55]: Seems like we need a custom hook or a utility for the function of set and remove localStorage.

File: src/components/Login/helpers/TokenService.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  firstName: string;
+  lastName: string;
+  email: string;
+  isActivated: boolean;
+  tokens: IUserToken;
+}
+class UserTokenService {
+  getUserInfo(): IUserInfo | undefined {
+    if (typeof window !== "undefined") {
+      const userInfo = JSON.parse(window.localStorage.getItem("UserInfo") as string);
```
Comment (Chanran) [inline src/components/Login/helpers/TokenService.ts:18]: Add "try catch" to wrap "JSON.parse" to handle getItem Error

File: src/components/Login/helpers/TokenService.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  userId: string;
+  firstName: string;
+  lastName: string;
+  email: string;
+  isActivated: boolean;
+  tokens: IUserToken;
+}
+class UserTokenService {
+  getUserInfo(): IUserInfo | undefined {
+    if (typeof window !== "undefined") {
```
Comment (Chanran) [inline src/components/Login/helpers/TokenService.ts:17]: Why need to judge "window" not to be "undefined"? all of browsers have window object

File: src/components/Login/helpers/TokenService.ts (hunk 1 above)
Comment (Chanran) [reply src/components/Login/helpers/TokenService.ts:55]: +1

File: src/components/Login/helpers/TokenService.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/* eslint-disable require-jsdoc */
```
Comment (Chanran) [inline src/components/Login/helpers/TokenService.ts:1]: @tatoMa This file seems to be a "common" or "utility". Is it correct that the file is placed in components folder？

Comment (jennawan) [reply src/components/Login/helpers/TokenService.ts]: Yes, the tokenservice may be required by other components. Moved to the util folder. Thx!

File: src/components/Login/helpers/TokenService.ts (hunk 3 above)
Comment (jennawan) [reply src/components/Login/helpers/TokenService.ts]: Removed unnecessary conditional statements.

File: src/components/Login/helpers/TokenService.ts (hunk 2 above)
Comment (jennawan) [reply src/components/Login/helpers/TokenService.ts]: Added try catch for handle getItem error.

File: src/components/Login/helpers/TokenService.ts (hunk 1 above)
Comment (jennawan) [reply src/components/Login/helpers/TokenService.ts]: Thanks for reminding. Add costom hook useHandleLocalStorageItem in hooks folder.

**Review by tatoMa — APPROVED** [review]
LGTM

File: src/components/Login/helpers/TokenService.ts (hunk 3 above)
Comment (tatoMa) [reply src/components/Login/helpers/TokenService.ts]: Just a reminder, we may need this window condition back later when we use SSR, as some of the codes may not run on the browser, but it is totally good for now. Just saying.

**Review by Chanran — APPROVED** [review]
LGTM

**Review by GLC-coder — APPROVED** [review]
LGTM

**Pull request merged**